training/DeepSpeed-Domino/domino/arguments.py

```diff
@@ -0,0 +1,400 @@
+        # bfloat16 requires gradient accumulation and all-reduce to
+        # be done in fp32.
+        if not args.accumulate_allreduce_grads_in_fp32:
+            args.accumulate_allreduce_grads_in_fp32 = True
+            if args.rank == 0:
+                print('accumulate and all-reduce gradients in fp32 for '
+                      'bfloat16 data type.', flush=True)
+
+    args.async_tensor_model_parallel_allreduce = True
+    args.gradient_accumulation_fusion = True
+    args.padded_vocab_size = 0 # tokenizer.py
+    args.model_type = 1
+    args.data_parallel_size = 1
+    args.DDP_impl = 'local'
+    args.use_contiguous_buffers_in_local_ddp = True
+    args.data_parallel_random_init = False
+
+    return args
+
+
+@dataclass
+class TransformerConfig():
+    """Configuration object for transformers.
+    """
+    sequence_parallel: bool = False
+    llama_model: bool = False
+    apply_residual_connection_post_layernorm = False
+    no_persist_layer_norm = False
+
+    # Initialization
+    perform_initialization: bool = True
+    use_cpu_initialization: bool = False
+
+    # Training
+    fp16: bool = False
+    bf16: bool = False
+    params_dtype: torch.dtype = torch.float32
+    timers: Callable = None
+
+    # Optimizations
+    gradient_accumulation_fusion: bool = True
+    async_tensor_model_parallel_allreduce: bool = True
+
+    # model architecture
+    num_layers: int = 0
+    hidden_size: int = 0
+    num_attention_heads: int = 0
+    ffn_hidden_size: int = None
+    kv_channels: int = None
+    hidden_dropout: float = 0.1
+    attention_dropout: float = 0.1
+    layernorm_epsilon: float = 1e-5
+    layernorm_zero_centered_gamma: bool = False
+    add_bias_linear: bool = True
+    swiglu = False
+    gated_linear_unit: bool = False
+    activation_func: Callable = F.gelu
+    bias_gelu_fusion = False
+
+    # initialization
+    init_method: Callable = None
+    output_layer_init_method: Callable = None
+    init_method_std: float = 0.02
+
+    enable_autocast: bool = False
+    # autocast_dtype: torch.dtype = None
+    deallocate_pipeline_outputs: bool = False
+    no_sync_func: Callable = None
+    # grad_sync_func: Callable = None
+    # param_sync_func: Callable = None
+
+    def __post_init__(self):
+        """ Python dataclass method that is used to modify attributes after initialization.
+            See https://docs.python.org/3/library/dataclasses.html#post-init-processing for more details.
+        """
+        if self.fp16 and self.bf16:
+            raise ValueError(
+                f'Only one of self.fp16: {self.fp16} and self.bf16 {self.bf16} should be True.'
+            )
+
+        # if self.num_attention_heads % self.tensor_model_parallel_size != 0:
+        #     raise ValueError(
+        #         f"num_attention_heads ({self.num_attention_heads}) must be a multiple of "
+        #         f"tensor_model_parallel_size ({self.tensor_model_parallel_size})."
+        #     )
```

Comment:
Should we remove the commented-out code?

Reply:
sure